### PR TITLE
Add plugins and providers to send event

### PR DESCRIPTION
**Pull request by markkaylor**
### What does it do?

Adds plugins and providers to 'didStartServer' send event

### CODE REVIEW

File: packages/strapi/lib/Strapi.js [hunk 1]
```diff
@@ -249,9 +249,24 @@ class Strapi {
...
-      await this.telemetry.send('didStartServer', { database: databaseClients });
+      // Get plugins
+      const plugins = Object.keys(this.plugins);
+      // Get providers
+      const providers = _(this.plugins)
+        .map(plugin => Object.keys(plugin.package.dependencies))
+        .flatten()
+        .filter(key => key.includes('strapi-provider-'))
+        .uniq()
+        .value();
```
Comment (markkaylor) [inline packages/strapi/lib/Strapi.js:262]: 🤷‍♂️

File: packages/strapi/lib/Strapi.js [hunk 2]
```diff
@@ -249,9 +249,33 @@ class Strapi {
...
 
-      // Emit started event.
+      // Get database clients
       const databaseClients = _.map(this.config.get('connections'), _.property('settings.client'));
-      await this.telemetry.send('didStartServer', { database: databaseClients });
+
+      // Get plugins
+      const plugins = { strapi: [], external: [] };
+      for (const plugin in this.plugins) {
+        if (this.plugins[plugin].package.author.name === 'Strapi team') {
```
Comment (alexandrebodin) [inline packages/strapi/lib/Strapi.js:258]: Checking hte package info is not going to be a safe info. 
When loading the plugins you should add a flag to say they are local or not but all strapi-plugin-xxx are external plugins even if not maintained by us. We should just ignore local plugins for now

File: packages/strapi/lib/Strapi.js [hunk 3]
```diff
@@ -249,9 +249,33 @@ class Strapi {
...
+          plugins.strapi.push(this.plugins[plugin].package.name);
+        } else {
+          plugins.external.push(this.plugins[plugin].package.name);
+        }
+      }
+
+      // Get providers
+      const providers = _(this.plugins)
```
Comment (alexandrebodin) [inline packages/strapi/lib/Strapi.js:266]: You should check the strapi-provider-xxx pacakges in strapi package json not for each plugin as this won't find manually installed providers. 

* Avoid using lodash chaining api

**Review by alexandrebodin — CHANGES_REQUESTED** [review]
We should avoid read package.json files

**Review by alexandrebodin — APPROVED** [review]
LGTM

**Pull request merged**